src/ImageRunModal.jsx

```diff
@@ -633,6 +588,82 @@ export class ImageRunModal extends React.Component {
         return owner === systemOwner;
     };
 
+    isFormValid = (validationFailed) => {
+        return Object.keys(validationFailed).length == 0;
+    };
+
+    async validateContainerName(containerName) {
+        try {
+            await client.containerExists(this.isSystem(), containerName);
+        } catch (error) {
+            return;
+        }
+        return _("Name already in use");
+    }
+
+    async validateForm() {
+        const { publish, volumes, env, containerName } = this.state;
+        const validationFailed = { };
+
+        const publishValidation = publish.map(a => {
+            let publishPortValidation;
+            publishPortValidation = validatePublishPort(a.IP, "IP", publishPortValidation);
+            publishPortValidation = validatePublishPort(a.hostPort, "hostPort", publishPortValidation);
+            publishPortValidation = validatePublishPort(a.containerPort, "containerPort", publishPortValidation);
+
+            return publishPortValidation;
+        });
+        if (publishValidation.some(entry => Object.keys(entry).length > 0))
+            validationFailed.publish = publishValidation;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231009-115441-607719f9-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)